modules/core/src/main/scala/org/scalasteward/core/update/PruningAlg.scala

```diff
@@ -64,10 +70,12 @@ final class PruningAlg[F[_]](implicit
       repoCache: RepoCache,
       dependencies: List[Scope.Dependency]
   ): F[(Boolean, List[Update.Single])] = {
-    val repoConfig = repoCache.maybeRepoConfig.getOrElse(RepoConfig.default)
+
     val depsWithoutResolvers = dependencies.map(_.value).distinct
     for {
       _ <- logger.info(s"Find updates for ${repo.show}")
+      defaultConfig <- repoConfigAlg.defaultRepoConfig
+      repoConfig = repoCache.maybeRepoConfig.getOrElse(defaultConfig)
```

Comment:
One last thing I noticed now: If `repoCache.maybeRepoConfig` is a `Some` there is no need to load the default repo config. We could change these two lines to:
```scala
repoConfig <- OptionT.fromOption[F](repoCache.maybeRepoConfig)
  .getOrElseF(repoConfigAlg.defaultRepoConfig)
```

Reply:
actually, in this case we should also merge configs
let me change